### ISSUE TITLE
rules_haskell buildkite broken 

### ISSUE THREAD

**Issue by c-parsons** [tweag/rules_haskell#1349]
It looks like rules_haskell is broken both HEAD and in-project, starting May 26.
Example Failure: https://buildkite.com/bazel/rules-haskell-haskell/builds/372#020c5972-dec5-4c7c-bff0-b4aa8cf2e372

Snip:
```
/usr/bin/ld.gold: error: /tmp/ghc3_0/ghc_2.o: requires dynamic R_X86_64_32 reloc against 'ZCMain_main_closure' which may overflow at runtime; recompile with -fPIC
--
  | /usr/bin/ld.gold: error: ../../../../../external/rules_haskell_ghc_linux_amd64/lib/base-4.13.0.0/libHSbase-4.13.0.0.a(Base.o): requires unsupported dynamic reloc 11; recompile with -fPIC
```

**Comment by aherrmann** [comment]
I've debugged this a bit further but haven't found a complete fix, yet.

I can reproduce the issue locally on Ubuntu 19.10 and on rules_haskell bindist CI when updating the image from Ubuntu 16.04 to Ubuntu 18.04, see 04e3b40d54c73d1c2aecee982917ef8e2f4efcee.

GHC has some logic around when to pass `-pie` or `-no-pie` and `-fPIC` or `-fno-PIC` that seems to be conflicting with more recent Ubuntu in the context of rules_haskell. I've tried passing various combinations of these flags to GHC. Passing [`-optl-no-pie`](https://github.com/tweag/rules_haskell/commit/02a886512c593263e727571fab1b40f8d4326eda) when linking executables fixed the issues on `haskell_binary` targets. However, there are still failures on `haskell_cabal_library|binary` targets.

**Comment by meteorcloudy** [comment]
Any progress on fixing this?
Still failing on Bazel CI: https://buildkite.com/bazel/rules-haskell-haskell/builds/399

**Comment by aherrmann** [comment]
Another small bit of progress. Undoing https://github.com/tweag/rules_haskell/pull/1325 avoids the issue, so this may be related to https://github.com/tweag/rules_haskell/blob/8cfef32ee9b14fc404a3bc4302955d17c2f88ce7/haskell/cc.bzl#L187-L190

I'll try to determine the required `cc` flags in the working setup using `strace` and see if I can fix this issue by passing them manually.

**Comment by aherrmann** [comment]
I've found a fix for this issue in #1388, I've tested in Bazel CI: https://buildkite.com/bazel/rules-haskell-haskell/builds/410
